components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/commons/js/dialog/sample-event-dialog.js

```diff
@@ -0,0 +1,149 @@
+                                    'value = "-1" disabled>-- Please Select a Stream name --</option>';
+                                var streamNames = self.generateOptions(data, initialOptionValue);
+                                sampleEventModal.find("select[name='stream-name']").html(streamNames);
+                            },
+                            function (data) {
+                                self.alertError("Error when retrieving siddhi streams. Reason: " + data);
+                            });
+                    });
+                    var $sampleEvent = sampleEventModal.find("textarea[name='generatedSampleEvent']");
+                    sampleEventModal.submit(function (event) {
+                        var eventType = sampleEventModal.find("select[name='event-format']").val();
+                        SampleEventRestClient.retrieveSampleEvent(
+                            sampleEventModal.find("select[name='siddhi-app-name']").val(),
+                            sampleEventModal.find("select[name='stream-name']").val(),
+                            eventType,
+                            function (data) {
+                                if (eventType === "xml") {
+                                    $sampleEvent.val(window.vkbeautify.xml(data));
+                                } else if (eventType === "json") {
+                                    $sampleEvent.val(window.vkbeautify.json(data));
+                                } else {
+                                    $sampleEvent.val(data);
+                                }
+                                $sampleEvent.select();
+                            },
+                            function (data) {
+                                self.alertError("Error when generating sample event. Reason: " + data.responseText);
+                            }
+                        );
+                        event.preventDefault();
+                    });
+                }
+            });
+
+        self.generateOptions = function (dataArray, initialOptionValue, componentName) {
+            var dataOption =
+                '<option value = "{{dataName}}">' +
+                '{{dataName}}' +
+                '</option>';
+            var result = '';
+            if (initialOptionValue !== undefined) {
+                result += initialOptionValue;
+            }
+            if (dataArray) {
+                dataArray.sort();
+                for (var i = 0; i < dataArray.length; i++) {
+                    if (componentName) {
+                        result += dataOption.replaceAll('{{dataName}}', dataArray[i][componentName]);
+                    } else {
+                        result += dataOption.replaceAll('{{dataName}}', dataArray[i]);
+                    }
+                }
+            }
+            return result;
+        };
+
+        self.alertError = function (errorMessage) {
```

Comment:
there is already a way we show alerts whether its error or warning in the editor ? this is a new one right ?

Reply:
@dnwick its the same thing that we have used in simulator also. the popup comes in top right corner. Is it changed?

Reply:
No i though we have a common place which we only need to pass the message. Now we have duplicate code for getErrorNotification .